### DISCUSSION TITLE
How can I use TipTap without importing it via modules?

### DISCUSSION THREAD

**Discussion by bjornhansen** [ueberdosis/tiptap#2074] [General Questions]
Is there a way to use it with just a good old fashioned <script> tag and and JS constructors? I don't want to set up a whole build routine with import/export.

**Comment by philippkuehn** [comment]
https://tiptap.dev/installation/cdn

**Reply by bjornhansen** [reply]
The JS on that page (below) uses `import` statements, which I'm hoping to avoid. Is there a way to use TipTap without them? E.g. under a global "TipTapEditor" constructor or something like that? Thanks for any insight!
```js
    import { Editor } from 'https://cdn.skypack.dev/@tiptap/core?min'
    import StarterKit from 'https://cdn.skypack.dev/@tiptap/starter-kit?min'
    const editor = new Editor({
      element: document.querySelector('.element'),
      extensions: [
        StarterKit,
      ],
      content: '<p>Hello World!</p>',
    })
```

**Reply by hanspagel** [reply]
Why would you avoid that? That example doesn’t even require any build setup.

**Reply by BrianLeishman** [reply]
I would like this because I am using typescript compiled with closure compiler, and there's no global object I can't tell closure compiler what to do correctly

**Reply by tkdaj** [reply]
> Why would you avoid that? That example doesn’t even require any build setup.

The example also reaches out to a separate URL which is blocked by strict CSP settings. I see no reason to limit the release of this lib to only allow its usage via JS modules.  I installed this into my project but it's an older .NET app with no JS build setup, so now I have to figure out how to make it work via regular script tags and probably an import map.

I also agree that one 100kB JS file is better than ten 10 kB files in some cases.  Modern browsers only allow a limited number (I think 6 or 8 max normally) of simultaneous requests, so each tiny request just blocks other requests from going out.

**Comment by robsontenorio** [comment]
@bjornhansen @philippkuehn 
did you solve it ?

@hanspagel 
Because he don’t want explicitly do that imports for dozen of modules. 

Old days = import a single “xxx.js” and itself provides a global class “TiptapEditor” with everything built in. 

I don’t see nothing wrong with a single 100kb JS file. I don’t like  of 10 files of 10kb approach. 

Devices nowadays are powerful. I  also would like to see  the “old days” approach.
